src/v/cloud_storage/anomalies_detector.cc

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "cloud_storage/anomalies_detector.h"
+
+#include "cloud_storage/base_manifest.h"
+#include "cloud_storage/partition_manifest.h"
+#include "cloud_storage/remote.h"
+#include "cloud_storage/spillover_manifest.h"
+
+namespace cloud_storage {
+
+anomalies_detector::anomalies_detector(
+  cloud_storage_clients::bucket_name bucket,
+  model::ntp ntp,
+  model::initial_revision_id initial_rev,
+  remote& remote,
+  retry_chain_logger& logger,
+  ss::abort_source& as)
+  : _bucket(std::move(bucket))
+  , _ntp(std::move(ntp))
+  , _initial_rev(initial_rev)
+  , _remote(remote)
+  , _logger(logger)
+  , _as(as) {}
+
+ss::future<anomalies_detector::result>
+anomalies_detector::run(retry_chain_node& rtc_node) {
+    anomalies detected{};
+    scrub_status status{scrub_status::full};
+    size_t ops = 0;
+
+    vlog(_logger.debug, "Downloading partition manifest ...");
+
+    partition_manifest manifest(_ntp, _initial_rev);
+    auto [dl_result, format] = co_await _remote.try_download_partition_manifest(
+      _bucket, manifest, rtc_node);
```

Comment:
Is it possible for this to run before the first manifest has been uploaded organically? I know the manifest upload interval is default 1min, but I recall there being some additional fudging of how we decide to upload

Reply:
In theory, yes. However, if that happens, then the trimming in `partition_manifest::process_anomalies` should take care of it.